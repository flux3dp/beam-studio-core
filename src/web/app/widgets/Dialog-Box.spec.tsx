import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DialogBox from './Dialog-Box';

test('should render correctly', () => {
  const mockOnClose = jest.fn();
  const wrapper = shallow(<DialogBox
    arrowDirection="top"
    arrowHeight={10}
    arrowWidth={20}
    arrowColor="black"
    arrowPadding={30}
    position={{}}
    onClose={mockOnClose}
    content="Hello World"
  />);
  expect(toJson(wrapper)).toMatchSnapshot();

  wrapper.find('.close-btn').simulate('click');
  expect(mockOnClose).toHaveBeenCalledTimes(1);
});
