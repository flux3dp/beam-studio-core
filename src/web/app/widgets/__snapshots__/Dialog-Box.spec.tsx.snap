// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="dialog-box-container"
  onClick={[MockFunction]}
  style={
    Object {
      "bottom": NaN,
      "right": NaN,
    }
  }
>
  <div
    className="dialog-box-arrow top"
    style={
      Object {
        "borderColor": "transparent transparent black transparent",
        "borderWidth": "0 10px 10px 10px",
        "right": 30,
      }
    }
  />
  <div
    className="dialog-box"
  >
    Hello World
  </div>
  <div
    className="close-btn right"
    onClick={[MockFunction]}
  >
    <div
      className="cross-wrapper"
    >
      <div
        className="bars bar1"
      />
      <div
        className="bars bar2"
      />
    </div>
  </div>
</div>
`;
